### PR TITLE
BUG: signal: fix crash in lfilter

### PR DESCRIPTION
**Pull request by pv**
Fixes gh-7991

### CODE REVIEW

**Review by larsoner — APPROVED** [review]
+1 for merge once the CIs come back happy

File: scipy/signal/lfilter.c.src [hunk 1]
```diff
@@ -193,9 +193,14 @@ scipy_signal_sigtools_linear_filter(PyObject * NPY_UNUSED(dummy), PyObject * arg
         if (str == NULL) {
             goto fail;
         }
+#if PY_VERSION_HEX >= 0x03000000
+        msg = PyUnicode_FromFormat(
+                        "input type '%U' not supported\n", str);
+#else
         s = PyString_AsString(str);
```
Comment (xoviat) [inline scipy/signal/lfilter.c.src]: Question: how would this compile on Python 3? `PyString_AsString` is not a function.

Comment (pv) [reply scipy/signal/lfilter.c.src]: there are compat defines

Comment (xoviat) [inline scipy/signal/lfilter.c.src]: This line is now completely useless. In fact, it will not compile.

**Review by ghost — APPROVED** [review]
Excellent job. I don't see any reason why the CI wouldn't pass.

**Pull request merged**